lib/networkx/multidigraph.rb

```diff
@@ -0,0 +1,233 @@
+module NetworkX
+  class MultiDiGraph < DiGraph
+    # Returns a new key
+    #
+    # @param node_1 [Object] the first node of a given edge
+    # @param node_2 [Object] the second node of a given edge
+    def new_edge_key(node_1, node_2)
+      return 0 if @adj[node_1][node_2].nil?
+      key = @adj[node_1][node_2].length
+      key += 1 while @adj[node_1][node_2].key?(key)
```

Comment:
Is this to handle deletion on intermediate key edges? If so, you can use maintain a `last_key` and use it here rather than `#length` and loop for coming to `last_key`.

Reply:
Then we would have to store the `last_key` attribute for every edge. Wouldn't that be an overhead? Also, we'd like to reuse the keys after deleting them.

Reply:
Also, I took the reference of networkx for the implementation of this method.